### PR TITLE
breaking: change app name and log path

### LINKED ISSUE #926: Change the log from `/tmp/cpeditorLogFiles` to `~/.cache/cpeditor/log`

### PR DESCRIPTION
**Pull request by ChungZH**
## Description

Change the log from `/tmp/cpeditorLogFiles` to `~/.cache/cpeditor/log`.

Use [QStandardPaths::](https://doc.qt.io/qt-6/qstandardpaths.html)CacheLocation.

Also change the appname from CP Editor to cpeditor.

## Related Issues / Pull Requests

close #926

## How Has This Been Tested?

Archlinux.
## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [ ] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [ ] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

### CODE REVIEW

**Review by neko-para — DISMISSED** [review]

**Comment by ouuan** [comment]
I think we can merge breaking changes into a v7.0 branch.

**Comment by coder3101** [comment]
Since the path `~/.cache/location/cpeditor/log` is only used by cpeditor and the location itself means that it should contain logs of cpeditor.

In the name of log file `cpeditor-*.log` is `cpeditor` prefix required anymore? 

We can better organise the log files by pushing them in a subdirectory by date. The log file can have a format `hh-mm-ss-tzz-instance.log` and we can delete the last 5 days or 10 days of log by subdirectory name instead of individual logs.

**Comment by coder3101** [comment]
> I think we can merge breaking changes into a v7.0 branch.

Agree. 

All PR targetting project "CP Editor 7.0" should go there.

**Comment by ouuan** [comment]
> Since the path `~/.cache/location/cpeditor/log` is only used by cpeditor and the location itself means that it should contain logs of cpeditor.
> 
> In the name of log file `cpeditor-*.log` is `cpeditor` prefix required anymore?

:+1:

> We can better organise the log files by pushing them in a subdirectory by date. The log file can have a format `hh-mm-ss-tzz-instance.log` and we can delete the last 5 days or 10 days of log by subdirectory name instead of individual logs.

I doubt whether this would be helpful :thinking: There aren't many log files so this can't significantly make it easier to navigate to a log file by date, but it does make it harder to navigate between log files. Neither of these two actions (find log file by date / navigate between log files) happens very often, though.

**Comment by ouuan** [comment]
And it's harder to keep 50 log files if they are in different folders. Then we can only keep log files in several days. However, if they are in the same folder, it is still easy to keep log files in several days.

**Comment by ChungZH** [comment]
Close #925 too?

**Comment by coder3101** [comment]
> And it's harder to keep 50 log files if they are in different folders. Then we can only keep log files in several days. However, if they are in the same folder, it is still easy to keep log files in several days.

I agree. We can keep it in the same top level directory but I still believe instead of deleting log files when it reaches 50, we should keep last 7 days of logs or 14 days.

If someone use app say 5 days a week, we end up with logs as old as 2 months. At the same time is we are developing cpeditor, in a a few days we can create 50 files.

Everytime we create new log file we delete all logs older than 7 days.

**Comment by coder3101** [comment]
> Close #925 too?

Not in the current code. If you wish you can do the changes and link that issue as well.

**Comment by ChungZH** [comment]
> > Close #925 too?
> 
> Not in the current code. If you wish you can do the changes and link that issue as well.

I'm afraid I'm too poor to do that 😹

**Comment by coder3101** [comment]
> > > Close #925 too?
> > 
> > Not in the current code. If you wish you can do the changes and link that issue as well.
> 
> I'm afraid I'm too poor to do that 😹 

Ok. Is changing name inside `SingleApplication::setApplicationName()` required for this log PR?

If not, undo that change and change title to `breaking: change log path and delete logs by date`

**Comment by neko-para** [comment]
I think it does require it 🤔 
As It influences, you have to change it you want it to work fine.

**Comment by ChungZH** [comment]
> > > > Close #925 too?也关闭925号？
> > > 
> > > 
> > > Not in the current code. If you wish you can do the changes and link that issue as well.当前代码中没有。如果您希望您可以进行更改并链接该问题。
> > 
> > 
> > I'm afraid I'm too poor to do that 😹恐怕我太穷了，做不到这一点
> 
> Ok. Is changing name inside `SingleApplication::setApplicationName()` required for this log PR?
> 
> If not, undo that change and change title to `breaking: change log path and delete logs by date`

image

If I don't change name, the log path will be `~/.cache/CP Editor/log`

**Comment by ChungZH** [comment]
I think the name can be changed in the future. Because we have to change the name in v7.

**Comment by neko-para** [comment]
Yes, maybe ignoring that is a good choice. That should be fixed in changing name pr.

**Comment by coder3101** [comment]
> Yes, maybe ignoring that is a good choice. That should be fixed in changing name pr.

Okay, is #925 waiting on some other PR? Can you propose a PR for it since you and ouuan handle settings related stuff. We can first merge your PR for #925 then merge this PR

**Comment by neko-para** [comment]
I don't think that #925 is blocked by any PR. @ouuan Do you know some infomation? The settings refactor branch handled by me isn't related to this.

**Comment by neko-para** [comment]
As this branch refers the app name by qt, it's ok to just merge this. There won't be a patch for this neither.

**Comment by ouuan** [comment]
> Okay, is #925 waiting on some other PR? Can you propose a PR for it since you and ouuan handle settings related stuff. We can first merge your PR for #925 then merge this PR

I don't understand. What does "an issue waits for a PR" mean?

**Comment by ouuan** [comment]
I don't see any problem to change the app name here. And I don't understand your conversation very well :thinking: What's the problem here? Are you discussing whether #925 should be closed by this or something else?

**Comment by coder3101** [comment]
> I don't see any problem to change the app name here. And I don't understand your conversation very well :thinking: What's the problem here? Are you discussing whether #925 should be closed by this or something else?

I am in favour of changing name here. OP say he cannot fix 925 in this PR but this PR requires that app name (Qt app name) should be changed. App name is used to find the cache directory.

**Comment by coder3101** [comment]
What we were discussing is that since this PR requires app name to be changed, instead of partially changing it, we can either

1. Completely fix #925 in this PR
2. Propose and merge a PR for #925 then merge this.

I think the change of app name should be done first in the v7.0 so that all following PR are created against new app name.

**Comment by coder3101** [comment]
@neko-para was asking what changes should done to fix #925?

**Comment by coder3101** [comment]
> a PR waiting for another PR/issue

It means PR A should be merged before PR B can be merged. Since PR B requires some changes that are in PR A.

If PR B is not opened, consider it's issue.

**Comment by ouuan** [comment]
OK, I understand the problem. The biggest problem is that without changes to the settings, the old settings won't be moved. We can wait for #858 to merge #934.

**Comment by coder3101** [comment]
Does #858 fixes #925? Or another PR to fix #925 will be created after #858 is merged?

**Comment by ouuan** [comment]
> Does #858 fixes #925?

Yes.

**Comment by coder3101** [comment]
> > Does #858 fixes #925?
> 
> Yes.

@neko-para you should link that issue to the PR.

**Pull request closed without merging**
